service/src/ocall_bridge/worker_on_chain_ocall.rs

```diff
@@ -141,6 +115,33 @@ where
 		}
 		status
 	}
+
+	fn send_confirmations(&self, confirmations: Vec<u8>) -> OCallBridgeResult<()> {
+		// TODO: improve error handling, using a mut status is not good design?
+		let mut status: OCallBridgeResult<()> = Ok(());
+		let api = self.node_api_factory.create_api();
+
+		// send confirmations to layer one
+		let confirmation_calls: Vec<OpaqueExtrinsic> =
+			match Decode::decode(&mut confirmations.as_slice()) {
+				Ok(calls) => calls,
+				Err(_) => {
+					status = Err(OCallBridgeError::SendBlockAndConfirmation(
+						"Could not decode confirmation calls".to_string(),
+					));
+					Default::default()
+				},
+			};
+
+		if !confirmation_calls.is_empty() {
+			debug!("Enclave wants to send {} extrinsics", confirmation_calls.len());
+			for call in confirmation_calls.into_iter() {
+				api.send_extrinsic(hex_encode(call.encode()), XtStatus::Ready).unwrap();
```

Comment:
I just saw this in the review here: We have some `unwrap`s still in this part o the code. We should probably fix this at some point?

Reply:
Yes, totally agreed! Does anything speak against fixing it in this PR? 😇

Reply:
Time is the reason 😝 For now we have to live with it I think, otherwise I will never be able to get through this refactoring in this sprint 😄 We have many places where we still use `unwrap()` instead of proper error handling. I think we should fix all of those in one go.